infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/StateTransfer.java

```diff
@@ -219,7 +220,9 @@ private static void transferChunk(Layout layout,
                 log.error("Missing address {} in batch {}", address, chunk);
                 throw new IllegalStateException("Missing address");
             }
-            entries.add((LogData) dataMap.get(address));
+            LogData ld = (LogData) dataMap.get(address);
+            ld.setPayloadCodecType(Codec.Type.valueOf(runtime.getParameters().getCodecType()));
```

Comment:
The logData read from server is in compressed state. If that logData has a different codec type than the state transfer runtime's codec type, then it would overwrite the logData's codec, which makes it inconsistent, right?

Reply:
I think you're correct. Cause we're not retrieving the payload, hence this is set before decompressing. Let me write a test for this case.